src/main/kotlin/no/skatteetaten/aurora/boober/feature/azure/AuroraAzureApimSubPart.kt

```diff
@@ -0,0 +1,88 @@
+package no.skatteetaten.aurora.boober.feature.azure
+
+import java.util.stream.Stream
+import com.fkorotkov.kubernetes.newObjectMeta
+import no.skatteetaten.aurora.boober.feature.name
+import no.skatteetaten.aurora.boober.feature.namespace
+import no.skatteetaten.aurora.boober.model.AuroraDeploymentSpec
+import no.skatteetaten.aurora.boober.model.AuroraResource
+import no.skatteetaten.aurora.boober.model.openshift.ApimSpec
+import no.skatteetaten.aurora.boober.model.openshift.AuroraApim
+import no.skatteetaten.aurora.boober.service.AuroraDeploymentSpecValidationException
+
+val AuroraDeploymentSpec.isApimEnabled: Boolean
+    get() {
+        return this.getOrNull(AuroraAzureApimSubPart.ConfigPath.enabled) ?: false
+    }
+
+val AuroraDeploymentSpec.azureApimPolicies: List<String>?
+    get() {
+        //  Somewhat strange construction in order to avoid error with null as typeCast
+        val rawGroup = this.getOrNull<Any>(AuroraAzureApimSubPart.ConfigPath.policies) ?: return null
+        return rawGroup as List<String>
+    }
+
+class AuroraAzureApimSubPart {
+    object ConfigPath {
+        private const val root = "azure/apim"
+        const val enabled = "$root/enabled"
+        const val path = "$root/path"
+        const val openapiUrl = "$root/openapiUrl"
+        const val serviceUrl = "$root/serviceUrl"
+        const val policies = "$root/policies"
+        const val apiHost = "$root/apiHost"
+    }
+
+    fun generate(adc: AuroraDeploymentSpec, azureFeature: AzureFeature): Set<AuroraResource> {
+        return if (adc.isApimEnabled) {
+            val path: String = adc[ConfigPath.path]
+            val openapiUrl: String = adc[ConfigPath.openapiUrl]
+            val serviceUrl: String = adc[ConfigPath.serviceUrl]
+            val apiHost: String = adc[ConfigPath.apiHost]
+
+            setOf(
+                azureFeature.generateResource(
+                    AuroraApim(
+                        _metadata = newObjectMeta {
+                            name = adc.name
+                            namespace = adc.namespace
+                        },
+                        spec = ApimSpec(
+                            path = path,
+                            openapiUrl = openapiUrl,
+                            serviceUrl = serviceUrl,
+                            policies = adc.azureApimPolicies!!,
```

Comment:
Kanskje dette er vanlig i Boober, men jeg synes det er bedre å kaste en exception som forklarer noe, om man virkelig vil feile her.

Altså

    adc.azureApimPolicies ?: throw SomeCleverException("explains it all")

Reply:
Dette er en vanlig konstruksjon i boober. Dersom du kommer hit i koden, har den validert, og adc.azureApimPolicies vil aldri være null.